### ISSUE TITLE
`filter` fails when filter just equals 1 value

### ISSUE THREAD

**Issue by zackarno** [r-tidy-remote-sensing/tidyrgee#19]
Here is reprex showing the issue:

``` r
library(rgee)
library(tidyrgee)


ee_Initialize()
#> -- rgee 1.1.2.9000 ---------------------------------- earthengine-api 0.1.295 -- 
#>  v user: not_defined
#>  v Initializing Google Earth Engine: v Initializing Google Earth Engine:  DONE!
#>  v Earth Engine account: users/zackarno 
#> --------------------------------------------------------------------------------
chirps_link <- "UCSB-CHG/CHIRPS/DAILY"
chirps <- ee$ImageCollection(chirps_link)

chirps_tidy <- as_tidyee(chirps)
chirps_tidy |> 
  filter(date=="2022-03-05")
#> Error in py_call_impl(callable, dots$args, dots$keywords): ee.ee_exception.EEException: Invalid argument specified for ee.List(): 15038
```

<sup>Created on 2022-06-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
  
Pretty sure this issue is cause by the fact that `ee$List()` cannot contains a vector of `length()==1`.  Below is the problematic code in the `filter.tidyee` method:

```r
ee_index_list <-  ee$List(vrt$tidyee_index |> as.character())
  ic_filt = .data$ee_ob$filter(ee$Filter$inList("tidyee_index", ee_index_list))
```

Should be an easy fix just to need to add something like (**caution:** below code not tested):

```
if(length(vrt$tidyee_index)==1){
  ee_index <- ee$String(vrt$tidyee_index |> as.character())
  ic_filt = .data$ee_ob$filter(ee$Filter$eq('tidyee_index', ee_index))
}

```
should add this fix ASAP as it is a pretty basic functionality missing

**Comment by zackarno** [comment]
issue should be fixed with commit 05e1302d9901548d4e41a77d2fdf5feb8a88b81e

**Comment by zackarno** [comment]
going to close for now as I think it is resolved
